data_structures/linked_list/doubly_linked_list.py

```diff
@@ -6,24 +6,21 @@
 - A Doubly Linked List (DLL) contains an extra pointer, typically called previous
     pointer, together with next pointer and data which are there in singly linked list.
  - Advantages over SLL - It can be traversed in both forward and backward direction.
-     Delete operation is more efficient"""
+     Delete operation is more efficient
+"""
 
 
-class LinkedList:
-    """
```

Comment:
Why remove all these tests?!?

Reply:
@cclauss I had removed the previous doc test as I had completely changed the implementation. Apologies for that
I have updated the doc test with a set of new test cases and added in much more cases
Kindly review the new changes submitted for the Dubly linked List

Some functions are intentionally made of O(N) time complexity to provide a more user-friendly output on the console.
besides most functions are working in O(1) time complexity

Kindly update in case of furthermore changes are required